backend/src/agent/persistence/worker/metrics_reporter.go

```diff
@@ -49,7 +49,11 @@ func (r MetricsReporter) ReportMetrics(workflow *util.Workflow) error {
 	if workflow.Status.Nodes == nil {
 		return nil
 	}
-	runID := string(workflow.UID)
+	if _, ok := workflow.ObjectMeta.Labels[util.LabelKeyWorkflowRunId]; !ok {
+		// Skip reporting if the workflow doesn't have the run id label
```

Comment:
When does this happen?

Reply:
It's unlikely going to happen. It only happen if someone delibrartly call this api with a workflow that's not created by KFP.